backend/app/core/embedded_celery.py

```
@@ -0,0 +1,164 @@
+    if _beat_instance is not None:
+        try:
+            logger.info("[EmbeddedCelery] Stopping beat scheduler gracefully...")
+            # Access the service (scheduler) from Beat instance
+            if hasattr(_beat_instance, "service") and _beat_instance.service:
+                scheduler = _beat_instance.service.scheduler
+                if scheduler and hasattr(scheduler, "lock"):
+                    # Release the RedBeat lock before shutdown
+                    try:
+                        scheduler.lock.release()
+                        logger.info("[EmbeddedCelery] Released RedBeat lock")
+                    except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)